realm/src/main/java/io/realm/internal/TableView.java

```diff
@@ -73,19 +73,24 @@
      * @param parent A table.
      * @param nativePtr pointer to table.
      */
-    protected TableView(Context context, Table parent, long nativePtr){
+    protected TableView(Context context, Table parent, long nativePtr) {
         this.context = context;
         this.parent = parent;
         this.nativePtr = nativePtr;
     }
 
+    protected TableView(Context context, Table parent, long nativePtr, TableQuery query) {
```

Comment:
No, the parent is the original table which the table view is fetching it data from. If you do a query on a table view, the new table view will still point back to the table. A statement like `r = r.where().findAll()` will create a table query (`r.where()`) which holds a native resource (a `realm::Query` object). When this table query is garbage collected, the native resource will deallocated, and the table view is rerun the query.

OK, my explanation isn't clear - and it calls for a drawing ;-)